### PR TITLE
Add support for accessing Secret Manager secret versions

### PR DESCRIPTION
**Pull request by owainhunt**
Adds support for hitting the [projects.secrets.versions.access](https://cloud.google.com/secret-manager/docs/reference/rest/v1/projects.secrets.versions/access) endpoint of the [Secret Manager REST API](https://cloud.google.com/secret-manager/docs/reference/rest). This is just one endpoint of a larger API, but is one of the most useful endpoints to hit from a server application.

### CODE REVIEW

**Comment by owainhunt** [comment]
cc @Andrewangeta

**Comment by Andrewangeta** [comment]
@owainhunt Looks good to me! @jordanebelanger did you wanna try to add the refreshing protection in so we could get all that plus this API out at once?

**Comment by owainhunt** [comment]
Just added some more error types. @Andrewangeta any progress on getting this merged in?

**Comment by Andrewangeta** [comment]
@owainhunt Sorry about getting this behind man 😭. I don't mind merging this in and then doing a separate release with improvements if you need this ASAP. Time is no where to be found but I will be on holiday starting next week and that should help give me time to catch up.

**Comment by owainhunt** [comment]
@Andrewangeta not a problem 🙂 we've been using the fork internally, and I'm on holiday now anyway, so no great rush if there are other things you want to get in with this one.

File: SecretManager/Sources/SecretManagerClient.swift [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import NIO
+
+public final class GoogleCloudSecretManagerClient {
+    
+    public var secrets: SecretsAPI
+    var secretManagerRequest: GoogleCloudSecretManagerRequest
+    
+    /// Initialize a client for interacting with the Google Cloud Secret Manager API
+    /// - Parameter credentials: The Credentials to use when authenticating with the APIs
+    /// - Parameter config: The configuration for the Cloud Datastore API
```
Comment (flopshot) [inline SecretManager/Sources/SecretManagerClient.swift]: typo: Cloud Datastore -> Cloud SecretManager

File: SecretManager/Sources/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+## Using the Secret Manager API
+
+### Setting up SecretManagerConfiguration
+
+To make GoogleCloudKit as flexible as possible to work with different API's and projects,
+you can configure each API with their own configuration if the default `GoogleCloudCredentialsConfiguration` doesn't satisfy your needs.
+
+For example the `GoogleCloudCredentialsConfiguration` can be configured with a `ProjectID`, but you might
+want to use this specific API with a different project than other APIs. Additionally every API has their own scope and you might want to configure.
+To use the Datastore API you can create a `GoogleCloudDatastoreConfiguration` in one of 2 ways.
```
Comment (flopshot) [inline SecretManager/Sources/README.md]: typo: Datastore -> SecretManager
typo: GoogleCloudDatastoreConfiguration -> GoogleCloudSecretManagerConfiguration

File: SecretManager/Sources/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+// OR
+let secretManagerConfiguration = GoogleCloudSecretManagerConfig.default() 
+// has full control access and uses default service account with no project specified.
+```
+
+### Now create a `GoogleCloudSecretManagerClient` with the configuration and an `HTTPClient`
+```swift
+let let client = HTTPClient(...)
+let smc = try GoogleCloudSecretManagerClient(credentials: credentialsConfiguration,
+                                       config: datastoreConfiguration,
```
Comment (flopshot) [inline SecretManager/Sources/README.md]: typo: datastoreConfiguration -> secretManagerConfiguration

File: SecretManager/Sources/README.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+For example the `GoogleCloudCredentialsConfiguration` can be configured with a `ProjectID`, but you might
+want to use this specific API with a different project than other APIs. Additionally every API has their own scope and you might want to configure.
+To use the Datastore API you can create a `GoogleCloudDatastoreConfiguration` in one of 2 ways.
+
+```swift
+let credentialsConfiguration = GoogleCloudCredentialsConfiguration(project: "my-project-1",
+                                                                   credentialsFile: "/path/to/service-account.json")
+
+let secretManagerConfiguration = GoogleCloudSecretManagerConfiguration(scope: [.datastore],
```
Comment (flopshot) [inline SecretManager/Sources/README.md]: typo: .datastore -> .cloudPlatform

**Comment by flopshot** [comment]
@owainhunt I tried your fork today and it works with my project perfectly.

**Comment by owainhunt** [comment]
@Andrewangeta any chance of getting this merged in?

**Comment by flopshot** [comment]
> @Andrewangeta any chance of getting this merged in?

I've been using this fork for a few months now. Still working as expected.

**Comment by Andrewangeta** [comment]
Yea I've been behind and this for a while now and let's get this merged in now and we'll do more significant changes like I was planning later on when time permits.

**Pull request merged**